### PR TITLE
[SPARK-38439][CORE][SQL] add curly braces with if, for and while statements

### PR DESCRIPTION
**Pull request by dcoliversun**
### What changes were proposed in this pull request?

Add braces with if,else,for,do and while statements.

### Why are the changes needed?

Braces are used with if, else, for, do and while statements, even if the body contains only a single statement. Avoid using the following example: 
```java
if (condition) statements;
```

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit test passed.

### CODE REVIEW

**Comment by dcoliversun** [comment]
@dongjoon-hyun @martin-g 
It would be good if you could take a look when you have time, thanks!

**Comment by HyukjinKwon** [comment]
Hm, I don't think this is worthwhile. Either way with/without braces is fine to me honestly.

**Comment by dcoliversun** [comment]
> Hm, I don't think this is worthwhile. Either way with/without braces is fine to me honestly.

@HyukjinKwon 
Hi. Code like the following is syntactically correct; but it is bug-prone
```java
if (condition) statements;
```
The problem is sooner or later you or someone else is going to need add a line to that if statement.
```java
if (condition) statements; statements;
```
The program has suddenly become buggy. Not everyone does when the buggy lines aren’t called out and pointed to as they are here. Even written like this, not everyone will see the bug immediately.

**Comment by yeikel** [comment]
Relevant : https://rules.sonarsource.com/java/RSPEC-121

**Comment by dcoliversun** [comment]
@cloud-fan It would be good if you could take a look when you have time, thanks!

**Comment by HyukjinKwon** [comment]
Not sure. Scala actually encourages to remove that curly braces (https://docs.scala-lang.org/style/control-structures.html). To me it still sounds like unnecessary

**Comment by yeikel** [comment]
> Not sure. Scala actually encourages to remove that curly braces (https://docs.scala-lang.org/style/control-structures.html). To me it still sounds like unnecessary 

Many of the modified files in this diff are .java files

**Comment by HyukjinKwon** [comment]
Java side actually follows Scala guide because Spark developers are used to Scala side:

> For Java code, Apache Spark follows [Oracle’s Java code conventions](http://www.oracle.com/technetwork/java/codeconvtoc-136057.html) and Scala guidelines below. The latter is preferred.

https://spark.apache.org/contributing.html

**Comment by HyukjinKwon** [comment]
In addition, I think these changes are rejected from other projects time to time, e.g.) `https://github.com/pi-hole/AdminLTE/issues/1027`

**Comment by dcoliversun** [comment]
@HyukjinKwon 
Thanks for your sharing. The information you provide can prove the value of my PR.

* [Oracle's Java code conventions](https://www.oracle.com/java/technologies/javase/codeconventions-contents.html)
image
image
image
* [Scala Control Structures](https://docs.scala-lang.org/style/control-structures.html)
image

It looks like this is necessary to add braces.

**Comment by dcoliversun** [comment]
@HyukjinKwon Hi. 

I remove curly braces from *if, else* statements. This PR only add curly braces with *if and while* statements, which follows Scala guide. 

I think it's good to use curly braces to prevent bugs. 

If you think it's unnecessary and should be rejected, I'm OK and close this PR. 😊

Thanks for your review.

**Comment by HyukjinKwon** [comment]
> The latter is preferred.

Scala guide is preferred.

**Comment by HyukjinKwon** [comment]
I don't mind if other committers like this change. So I will defer to them.

**Comment by dcoliversun** [comment]
Sorry to bother you @HyukjinKwon @dongjoon-hyun @martin-g 
And thanks for your work @HyukjinKwon @yeikel 
This PR is useless and close it.

File: common/network-common/src/main/java/org/apache/spark/network/util/LevelDBProvider.java [hunk 1]
```diff
@@ -134,8 +134,12 @@ public StoreVersion(@JsonProperty("major") int major, @JsonProperty("minor") int
 
     @Override
     public boolean equals(Object o) {
-      if (this == o) return true;
```
Comment (cloud-fan) [inline common/network-common/src/main/java/org/apache/spark/network/util/LevelDBProvider.java:137]: you know what, these code are generated by IntelliJ...

Comment (dcoliversun) [reply common/network-common/src/main/java/org/apache/spark/network/util/LevelDBProvider.java:137]: I don't know about these. Sorry to bother you too. Please ignore this PR

**Pull request closed without merging**